packages/backend/src/core/tx-count/BlockTxCountUpdater.ts

```diff
@@ -0,0 +1,85 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectId, UnixTime } from '@l2beat/types'
+
+import { TxCountRepository } from '../../peripherals/database/TxCountRepository'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { Clock } from '../Clock'
+
+export class BlockTxCountUpdater {
```

Comment:
I think that instead of this being called `BlockTxCountUpdater` it should be called `RpcTransactionUpdater` as most networks have blocks but not all of them use the standard ethereum rpc.